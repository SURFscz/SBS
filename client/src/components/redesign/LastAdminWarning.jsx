import React from "react";
import "./LastAdminWarning.scss";
import {ReactComponent as CriticalIcon} from "../../icons/critical.svg";
import I18n from "i18n-js";

export default function LastAdminWarning({organisation, currentUserDeleted = true}) {
    const msg = I18n.t(`collaborationDetail.${currentUserDeleted ? "lastAdminWarningUser" : "lastAdminWarning"}`,
        {name: organisation.name});
    return (
        <span className="last-admin-error-indication"><CriticalIcon/>
            <h2 dangerouslySetInnerHTML={{__html: msg}}/>
        </span>);

}