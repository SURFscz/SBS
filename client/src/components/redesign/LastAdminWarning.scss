@import "../../stylesheets/vars";

span.last-admin-error-indication {
    color: $error-red;
    flex-direction: column;
    display: flex;
    align-items: center;

    svg {
        margin-bottom: 15px;
        width: 40px;
        height: auto;
    }
}