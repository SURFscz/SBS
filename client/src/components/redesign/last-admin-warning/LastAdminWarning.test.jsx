import React from "react";
import { render } from "@testing-library/react";
import LastAdminWarning from "./LastAdminWarning";

describe("LastAdminWarning", () => {
    it("renders with default props", () => {
        const { asFragment } = render(
            <LastAdminWarning
                organisation={{ name: "Test Organisation" }}
                currentUserDeleted={true}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders with currentUserDeleted false", () => {
        const { asFragment } = render(
            <LastAdminWarning
                organisation={{ name: "Test Organisation" }}
                currentUserDeleted={false}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders with custom localePrefix", () => {
        const { asFragment } = render(
            <LastAdminWarning
                organisation={{ name: "Test Organisation" }}
                currentUserDeleted={true}
                localePrefix="customLocale"
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
