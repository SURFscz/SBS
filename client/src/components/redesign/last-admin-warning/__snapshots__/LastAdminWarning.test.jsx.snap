// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LastAdminWarning renders with currentUserDeleted false 1`] = `
<DocumentFragment>
  <p
    class="last-admin-error-indication"
  >
    <span>
      You are leaving this collaboration without admins. After this, management of the collaboration can only be carried out by the admins of organisation Test Organisation
    </span>
  </p>
</DocumentFragment>
`;

exports[`LastAdminWarning renders with custom localePrefix 1`] = `
<DocumentFragment>
  <p
    class="last-admin-error-indication"
  >
    <span>
      [missing "customLocale.lastAdminWarningUser" translation]
    </span>
  </p>
</DocumentFragment>
`;

exports[`LastAdminWarning renders with default props 1`] = `
<DocumentFragment>
  <p
    class="last-admin-error-indication"
  >
    <span>
      You are the last admin in this collaboration. After you leave, management of the collaboration can only be carried out by the admins of organisation Test Organisation
    </span>
  </p>
</DocumentFragment>
`;
